tasks/main.yml

```diff
@@ -17,10 +17,21 @@
     - tinc_requirements
     - install
 
+# the package module for opkg does not support state=latest
 - name: install tinc
   package:
     name: tinc
     state: latest
+  when: ansible_pkg_mgr != "opkg"
+  tags:
+    - tinc_install
+    - install
+
+- name: install tinc
+  package:
+    name: tinc
+    state: present
+  when: ansible_pkg_mgr == "opkg"
```

Comment:
Likewise, this could be summarized into a single task, or even allow the user to define it, by adding in defaults/main.yml

``tinc_package_state: present``

It's not a big deal for me if by default it's present instead of absent.

But on top of that we could do the following:

``
package:
  name: tinc
  state: "{{ (tinc_package_state == 'latest' and ansible_pkg_mgr == 'opkg') | ternary('present', tinc_package_state) }}
``

Reply:
I like the idea of using the variable in the defaults with present.  More versatile.  I'll update.